sqlglot/dialects/presto.py

```diff
@@ -153,6 +153,20 @@ def _unnest_sequence(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def _first_last_sql(self: generator.Generator, expression: exp.First | exp.Last) -> str:
+    """
+    Trino doesn't support FIRST / LAST as functions, but they're valid in the context
+    of MATCH_RECOGNIZE, so we need to preserve them in that case. In all other cases
+    they're converted into an ARBITRARY call.
+
+    Reference: https://trino.io/docs/current/sql/match-recognize.html#logical-navigation-functions
+    """
+    if expression.find_ancestor(exp.MatchRecognize):
```

Comment:
you don't want this to bubble up all the way, so maybe like isinstance(expression.find_ancestor(exp.MatchRecognize, exp.Select), exp.MatchRecognize)